src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+
+    /**
+     * Returns corresponding command specified in the input string.
+     * @param str The input string.
+     * @return Command.
+     * @throws DukeException If the input is incomplete.
+     */
+    public static Command parse(String str) throws DukeException {
+        String[] split1 = str.split(" ", 2);
+        String commandWord = split1[0].toLowerCase();
+        if (split1.length < 2 || split1[1].equals("")) {
+            // one word command
+            if (commandWord.equals("list")) {
+                return new ListCommand();
+            } else if (commandWord.equals("bye")) {
+                return new ExitCommand();
+            } else if (commandWord.equals("done")) {
+                throw new DukeException("Please tell me the task number that you have completed.");
+            } else if (commandWord.equals("date")) {
+                throw new DukeException("Please provide a date.");
+            } else if (commandWord.equals("delete")) {
+                throw new DukeException("Please tell me the task number that you want to delete.");
+            } else if (commandWord.equals("todo")) {
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            } else if (commandWord.equals("deadline")) {
+                throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+            } else if (commandWord.equals("event")) {
+                throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+            } else {
+                throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } else {
+            // more than one word command
+            String commandBody = split1[1].trim();
+            if (commandWord.equals("done")) {
+                // create done command: "done 1"
+                return new DoneCommand(Integer.parseInt(commandBody));
+            } else if (commandWord.equals("delete")) {
+                // delete 2
+                return new DeleteCommand(Integer.parseInt(commandBody));
+            } else if (commandWord.equals("date")) {
+                // create date command: "date 2020-09-24"
+                LocalDate time = LocalDate.parse(commandBody);
+                return new DateCommand(time);
+            } else if (commandWord.equals("find")) {
+                return new FindCommand(commandBody);
+            } else {
+                // add command
+                Task newTask = Parser.parseTask(str);
+                return new AddCommand(newTask);
+            }
+        }
+    }
+
+    /**
+     * Returns a task corresponding to the task string from storage file.
+     * @param taskStr The task string from storage file.
+     * @return Task.
+     */
+    public static Task parseTaskFromStorage(String taskStr) {
+        Task savedTask;
+        String[] split = taskStr.split(Pattern.quote(" | "));
```

Comment:
I think this variable name "split" here and some other places below might could be changed into a more meaningful name, too.